python/ingestion/cdc_wonder_utils.py

```diff
@@ -0,0 +1,263 @@
+from ingestion.het_types import GEO_TYPE, CANCER_TYPE_OR_ALL
+from ingestion import gcs_to_bq_util, dataset_utils
+import ingestion.standardized_columns as std_col
+from ingestion.constants import CURRENT, STATE_LEVEL, NATIONAL_LEVEL, US_FIPS
+import pandas as pd
+import numpy as np
+from typing import Dict, cast, List
+from ingestion.merge_utils import merge_dfs_list
+from ingestion.dataset_utils import combine_race_ethnicity
+
+# CDC Wonder Constants
+YEAR_COL = "Year"
+RACE_NAME = "Race"
+AGE_GROUP = "Age Groups"
+SEX_COL = "Sex"
+COUNT_COL = "Count"
+POP_COL = "Population"
+CRUDE_RATE_COL = "Crude Rate"
+
+# State column names for different breakdowns
+STATE_CODE_RACE = "States Code"
+STATE_CODE_DEFAULT = 'States and Puerto Rico Code'
+
+TMP_ALL: CANCER_TYPE_OR_ALL = 'all'
+CDC_WONDER_DIR = 'cdc_wonder'
+
+# Cancer conditions based on sex breakdown requirements
+CANCERS_WITH_SEX_BREAKDOWN = ["Colorectal", "Lung"]
+ALL_CANCER_CONDITIONS = ["Breast", "Cervical", "Prostate"] + CANCERS_WITH_SEX_BREAKDOWN
+
+# Mapping for demographic breakdowns
+breakdown_het_to_source_type = {
+    "age": AGE_GROUP,
+    "race_and_ethnicity": RACE_NAME,
+    "sex": SEX_COL,
+}
+
+BREAKDOWN_TO_STANDARD_BY_COL = {
+    # Age source groups already match needed HET groups
+    std_col.RACE_CATEGORY_ID_COL: {
+        'American Indian or Alaska Native': std_col.Race.AIAN_NH.value,
+        'Asian or Pacific Islander': std_col.Race.API_NH.value,
+        'Hispanic': std_col.Race.HISP.value,
+        'White': std_col.Race.WHITE_NH.value,
+        'Black or African American': std_col.Race.BLACK_NH.value,
+        'Other Races and Unknown combined': std_col.Race.OTHER_NONSTANDARD_NH.value,
+    },
+    # Sex source groups already match needed HET groups
+}
+
+DTYPE = {STATE_CODE_RACE: str, STATE_CODE_DEFAULT: str}
+
+
+def get_state_code_col(breakdown: CANCER_TYPE_OR_ALL) -> str:
+    """Returns appropriate state code column based on breakdown type and geo level"""
+    if breakdown == std_col.RACE_OR_HISPANIC_COL:
+        return STATE_CODE_RACE
+    return STATE_CODE_DEFAULT
+
+
+def get_sheet_name(geo_level: GEO_TYPE, breakdown: CANCER_TYPE_OR_ALL) -> str:
+    """geo_level: string equal to `national`, or `state`
+    breakdown: string demographic breakdown type
+    return: a string sheet name based on the provided args"""

```

Comment:
nice using the tuple as the dict key; love that python can do that. can we not use the term `sheet` though? it makes me think we are reading from an excel file with sub sheets. maybe just like get_source_breakdown or something. 

Also in general, I'd love to get ALL our datasource away from using the term `breakdown` when we really mean `demographic_yype`. I think "breakdown" should me the exact demo, geo, and time breakdown of a dataset. Can rename now or we can do a big PR to rename all the data sources at once